### PR TITLE
Deal with a zero rate SSPoisson case correctly

### PR DESCRIPTION
**Pull request by andrewgait**
Fix for issue https://github.com/SpiNNakerManchester/sPyNNaker/issues/589 - this should allow rate=0 in all cases.

### CODE REVIEW

File: spynnaker/pyNN/models/neuron/synaptic_manager.py [hunk 1]
```diff
@@ -461,7 +461,17 @@ def _get_ring_buffer_to_input_left_shifts(
                     spikes_per_second = self._spikes_per_second
                     if isinstance(app_edge.pre_vertex,
                                   SpikeSourcePoissonVertex):
-                        spikes_per_second = app_edge.pre_vertex.rate
+                        rate = app_edge.pre_vertex.rate
+                        # If non-zero rate then use it; otherwise keep default
+                        if hasattr(rate, "__getitem__"):
+                            sumrate = 0
+                            for i in range(len(rate)):
```
Comment (rowleya) [inline spynnaker/pyNN/models/neuron/synaptic_manager.py]: probably easier to do sumrate = sum(rate)

Comment (andrewgait) [reply spynnaker/pyNN/models/neuron/synaptic_manager.py]: Definitely is - fixed.

File: spynnaker/pyNN/models/neuron/synaptic_manager.py [hunk 2]
```diff
@@ -461,7 +461,10 @@ def _get_ring_buffer_to_input_left_shifts(
                     spikes_per_second = self._spikes_per_second
                     if isinstance(app_edge.pre_vertex,
                                   SpikeSourcePoissonVertex):
-                        spikes_per_second = app_edge.pre_vertex.rate
+                        rate = app_edge.pre_vertex.rate
+                        # If non-zero rate then use it; otherwise keep default
+                        if (sum(rate) != 0):
```
Comment (rowleya) [inline spynnaker/pyNN/models/neuron/synaptic_manager.py]: Almost - you need some of what you had before, as you can only sum things that are lists (unless this is always guaranteed to be a list).

Comment (andrewgait) [reply spynnaker/pyNN/models/neuron/synaptic_manager.py]: It's guaranteed to be a list (SpikeSourcePoissonVertex.rate is set using utility_calls.convert_param_to_numpy, which enforces that single values become single-valued arrays).

**Comment by rowleya** [comment]
Updated to account for new information available.  I have tested it against the case in the issue and this seems to be fine.  I will merge when green...

**Pull request merged**